lib/appenders/dateFile.js

```diff
@@ -22,17 +22,26 @@ function appender(
   // options should work for dateFile as well.
   options.maxSize = options.maxLogSize;
 
+  let alive = true;
+
   const logFile = new streams.DateRollingFileStream(
     filename,
     pattern,
     options
   );
 
+  logFile.on('error', (err) => {
```

Comment:
> If an `EventEmitter` does not have at least one listener registered for the `'error'` event, and an `'error'` event is emitted, the error is thrown, a stack trace is printed, and the Node.js process exits.
> _(src: https://nodejs.org/docs/latest/api/events.html#error-events)_

Similar to #529 and #1089, the dateFileAppender causes the the Node.js process to exit as there is no listener for its `'error'` event. 
This PR by @4eb0da added the listener to prevent such crashes.